trajopt/CMakeLists.txt

```diff
@@ -45,6 +45,7 @@ if(CXX_FEATURE_FOUND EQUAL "-1")
 else()
     target_compile_features(${PROJECT_NAME} PUBLIC cxx_std_11)
 endif()
+target_compile_options(${PROJECT_NAME} PUBLIC -mno-avx)
```

Comment:
Does this work with compilers other than GCC?

Reply:
It looks like this may work with Clang and AppleClang per [VTK](https://gitlab.kitware.com/vtk/vtk-m/commit/514ea09afcf6be56f887c96858d55a9a66fb37cb) but not all compilers. I did look and does no appear to be a feature so probably need to do something similar to what VTK did in the like. @gavanderhoorn do you know of any other solutions?

Reply:
No, not really.

I'm not sure we'd need something as elaborate as the VTK implementation, but if you need to set these compiler options across many different packages something which avoids duplication would be good.

We could compromise and only 'support' GCC for now by wrapping these `target_compile_options(..)` in a:

```cmake
if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
endif()
```

Reply:
@gavanderhoorn We updated it with your suggestion and added a warning if not using GNU.